src/borrow_tracker/tree_borrows/perms.rs

```diff
@@ -634,6 +663,33 @@ mod propagation_optimization_checks {
         }
     }
 
+    #[test]
+    #[rustfmt::skip]
+    fn permission_sifa_is_correct() {
+        // Tests that `strongest_idempotent_foreign_access` is correct. See `foreign_access_skipping.rs`.
+        for perm in PermissionPriv::exhaustive() {
+            // Assert that adding a protector makes is less idempotent.
+            if perm.compatible_with_protector() {
+                assert!(perm.strongest_idempotent_foreign_access(true) <= perm.strongest_idempotent_foreign_access(false));
+            }
+            for prot in bool::exhaustive() {
+                if prot {
+                    precondition!(perm.compatible_with_protector());
+                }
+                let access = perm.strongest_idempotent_foreign_access(prot);
+                // We now assert it is idempotent, and never causes UB.
+                // First, if the SIFA includes foreign reads, assert it is idempotent under foreign reads.
+                if LastForeignAccess::ForeignRead <= access {
+                    assert_eq!(perm, transition::perform_access(AccessKind::Read, AccessRelatedness::DistantAccess, perm, prot).unwrap());
```

Comment:
What's a "DistantAccess"? Is that the same as "foreign"? If yes, we should rename. (Can be a separate PR.)

Reply:
no, it's a cousin access. The `AccessRelatedness` enum has 4 constructors (child, self, cousin, parent), but for almost all cases, two of them (child+self, cousin+parent) are treated the same. The only exception is in protector end writes.

But here we need to choose a concrete inhabitant. We could also pick `Parent`, i.e. a different member of the same equivalence class.

Reply:
Ah, we should rename to `CousinAccess` then.

And please add a comment here explaining what you just said.